lib/SILOptimizer/Analysis/ARCAnalysis.cpp

```diff
@@ -1154,3 +1156,132 @@ SILInstruction *swift::findReleaseToMatchUnsafeGuaranteedValue(
   }
   return nullptr;
 }
+
+void EpilogueARCContext::initializeDataflow() {
+  // Find the exit blocks.
+  auto Throw = F->findThrowBB();
+  auto Return = F->findReturnBB();
+  if (Throw != F->end())
+    ExitBlocks.insert(&*Throw);
+  if (Return != F->end())
+    ExitBlocks.insert(&*Return);
+  // Return and throw blocks BBSetOut is initialized to true.
+  for (auto &B : *F) {
+    EpilogueARCBlockStates[&B] =
+              new (BPA.Allocate()) EpilogueARCBlockState();
```

Comment:
Using findThrowBB, findReturnBB is not efficient here because those functions iterate over all blocks.
Instead you could just add the following into this loop:
if (B.getTerminator()->isFunctionExiting) { ExitBlocks.insert(&B); }